src/integration/cloneDocuSources.js

```diff
@@ -56,5 +61,20 @@ function cloneDocuRepo(topicDetails, cb) {
   });
 }
 
-
 module.exports = cloneDocuSources;
+
+/**
+ * This function creates a JSON object in which names of not cloned repositories are stored.
+ * @param {Object} [topicDetails] - holds details of the repo that needs to be cloned.
+ * @param {Function} [cb] - callback for asynchronous operation
+ */
+function _createMatrixWithNotClonedRepositories(topicDetails, config, cb) {
+
+  //push to array all not cloned repositories
+  notClonedArray.push(topicDetails.clonedGenRNDestLocation, topicDetails.clonedGenRNDestLocationInternal, topicDetails.clonedGenDestLocation, topicDetails.clonedGenDestLocationInternal);
+
+  //write array the file
+  creator.createFile(`${config.tempLocation}/notClonedRepositories.json`, notClonedArray, (err) => {
+  });
```

Comment:
add some error handling (just a message that something went wrong with writting matrix file)